contrib/src/main/java/com/datatorrent/contrib/parser/LogParser.java

```diff
@@ -0,0 +1,236 @@
+ * under the License.
+ */
+package com.datatorrent.contrib.parser;
+
+import java.io.IOException;
+import javax.validation.constraints.NotNull;
+import org.codehaus.jettison.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.commons.lang3.CharEncoding;
+import org.apache.commons.lang3.StringUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.lib.parser.Parser;
+import com.datatorrent.lib.util.KeyValPair;
+
+/**
+ * Operator that parses a log string tuple against the
+ * a specified json schema and emits POJO on a parsed port and tuples that could not be
+ * parsed on error port.<br>
+ * <b>Properties</b><br>
+ * <b>jsonSchema</b>:schema as a string<br>
+ * <b>clazz</b>:Pojo class in case of user specified schema<br>
+ * <b>Ports</b> <br>
+ * <b>in</b>:input tuple as a String. Each tuple represents a log<br>
+ * <b>parsedOutput</b>:tuples that are validated against the specified schema are emitted
+ * as POJO on this port<br>
+ * <b>err</b>:tuples that do not confine to log format are emitted on this port as
+ * KeyValPair<String,String><br>
+ * Key being the tuple and Val being the reason.
+ */
+public class LogParser extends Parser<byte[], KeyValPair<String, String>>
+{
+  private transient Class<?> clazz;
+
+  @NotNull
+  private String logFileFormat;
+
+  private String encoding;
+
+  private LogSchemaDetails logSchemaDetails;
+
+  private transient ObjectMapper objMapper;
+
+  @Override
+  public Object convert(byte[] tuple)
+  {
+    throw new UnsupportedOperationException("Not supported");
+  }
+
+  @Override
+  public KeyValPair<String, String> processErrorTuple(byte[] bytes)
+  {
+    return null;
+  }
+
+  /**
+   * output port to emit valid records as POJO
+   */
+  public transient DefaultOutputPort<Object> parsedOutput = new DefaultOutputPort<Object>()
+  {
+    public void setup(Context.PortContext context)
+    {
+      clazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+  /**
+   * metric to keep count of number of tuples emitted on {@link #parsedOutput}
+   * port
+   */
+  @AutoMetric
+  long parsedOutputCount;
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    parsedOutputCount = 0;
+  }
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    objMapper = new ObjectMapper();
+    setupLog();
+  }
+
+  @Override
+  public void processTuple(byte[] inputTuple)
+  {
+    if (inputTuple == null) {
+      this.admitError(null, "null tuple");
+      return;
+    }
+    String incomingString = "";
+    try {
+      if (encoding != null) {
+        incomingString = new String(inputTuple, encoding);
+      } else {
+        incomingString = new String(inputTuple, CharEncoding.UTF_8);
+      }
```

Comment:
A little more compact:
{code}
String enc = encoding != null ? encoding : CharEncoding.UTF_8;
incomingString = new String(inputTuple, encoding);
{code}

Actually, if the encoding is not expected to change midstream,  you can check the field and set
it in setup() -- then you can avoid the test for every tuple here.